### PR TITLE
Update ec2 discovery property name

### PR DESCRIPTION
**Pull request by bploetz**
Per discussion in Slack, the documented `discovery.type: ec2` config property causes the following warning in the logs:

```
[2018-03-06T15:01:46,933][WARN ][o.e.d.d.e.Ec2DiscoveryPlugin] using [discovery.type] setting to set hosts provider is deprecated; set [discovery.zen.hosts_provider: ec2] instead
```

Changing this property to the new `discovery.zen.hosts_provider: ec2` removes this warning.

### CODE REVIEW

File: docs/deployment/cloud/aws/ec2-setup.txt [hunk 1]
```diff
@@ -37,9 +37,9 @@ will allow CrateDB to accept and respond to pings from other CrateDB instances
 with the same cluster name and form a cluster.
 
 Once you have your instances running and CrateDB installed, you can enable the
-EC2 discovery by setting the discovery type to ``ec2``::
+EC2 discovery by setting the discovery.zen.hosts_provider to ``ec2``::
```
Comment (nslater) [inline docs/deployment/cloud/aws/ec2-setup.txt]: please rephrase this as:

> by setting \`\`discovery.zen.hosts_provider\`\` to \`\`ec2\`\`

**Comment by nslater** [comment]
thank you for the pull request, @bploetz! I've requested a minor change to make sure the sentence reads properly

also, before we merge this, could you sign our [CLA](https://crate.io/community/contribute/cla/)? I realise this is a very small change, so sorry for the additional hassle

**Comment by bploetz** [comment]
@nslater I made the requested change and signed the CLA. Let me know if you need anything else.

**Comment by nslater** [comment]
this is great! thanks so much. can you ``git rebase -i origin/master` and mark your second commit as a fixup and do a ``git push -f``? then I'll merge! :)

**Comment by bploetz** [comment]
@nslater all set

**Comment by nslater** [comment]
@bploetz merged! thank you very much!
